prime-router/src/main/kotlin/cli/ProcessHl7Commands.kt

```diff
@@ -0,0 +1,236 @@
+            val children = output.getAll(iname)
+            children.forEachIndexed { index, c ->
+                indexStructure(c, (index + 1).toString(), outputMap)
+            }
+        }
+
+        fun compareHl7Type(
+            segmentIndex: String,
+            input: Type,
+            output: Type,
+            segmentType: String,
+            segmentNumber: Int
+        ): Boolean {
+            return when {
+                input is Primitive && output is Primitive && !StringUtils.equals(input.value, output.value) -> {
+                    echo(
+                        "$segmentIndex-$segmentType number $segmentNumber did not match on ${input.value}, " +
+                            output.value
+                    )
+                    return false
+                }
+
+                input is Varies && output is Varies -> compareHl7Type(
+                    segmentIndex,
+                    input.data,
+                    output.data,
+                    segmentType,
+                    segmentNumber
+                )
+
+                input is Composite && output is Composite -> {
+                    val inputComponents = input.components.filter { !it.isEmpty }
+                    val outputComponents = output.components.filter { !it.isEmpty }
+                    val inputExtraComponents = input.extraComponents
+                    val outputExtraComponents = output.extraComponents
+                    if (inputComponents.size != outputComponents.size) {
+                        echo("$segmentIndex-$segmentType number $segmentNumber did not match")
+                        return false
+                    } else if (inputExtraComponents.numComponents() != outputExtraComponents.numComponents()) {
+                        echo("$segmentIndex-$segmentType number $segmentNumber did not match")
+                        return false
+                    } else {
+                        inputComponents.zip(outputComponents).forEach { (i, o) ->
+                            if (!compareHl7Type(segmentIndex, i, o, segmentType, segmentNumber)) {
+                                echo("$segmentIndex-$segmentType number $segmentNumber did not match on $i, $o")
+                                return false
+                            }
+                        }
+                        return true
+                    }
+                }
+
+                else -> {
+                    return true
+                }
+            }
+        }
+
+        val mapNumOfSegment = mutableMapOf<String, Int>()
+
+        inputMap.forEach { (segmentIndex, segment) ->
+            // used to get the name of the segment as well as which number of that segment it is
+            val segmentIndexParts = segmentIndex.split("-")
+            val segmentType = segmentIndexParts[segmentIndexParts.size - 1]
+            val num = mapNumOfSegment[segmentType]
```

Comment:
I found the other way of mapping confusing/complicated, so I wanted a simple way where I didn't have to count multiple types of segments to figure out which one the message pertains to. This will allow them to get the number of just that segment type so they just have to count those segments instead, if they prefer. So, it is the overall number, not the number under the observation result.

Reply:
Let's pair on this on Monday, but this is saying that if you have:
```
OBR
OBX
OBR
OBX
SPM
OBX
```

and the last OBX doesn't match, the output would say that 3rd OBX is wrong?